src/main/java/seedu/address/logic/commands/consultation/AddConsultCommand.java

```diff
@@ -54,6 +54,14 @@ public CommandType getCommandType() {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+
+        // Check if the model already has a consultation with the same date and time
+        if (model.hasConsult(newConsult)) {
+            throw new CommandException("Duplicate consultation."
```

Comment:
Maybe consider adding a testcase for this? Should be relatively simple, just add the same consultation twice and check that it causes an error.